### PR TITLE
port to protocol-9p >= 0.11.0

### PR DESCRIPTION
**Pull request by avsm**
- several packages renamed into jbuilder-style module namespaces
- use upstream `send_pings` support for 9P heartbeats

w/ assist from @djs55

### CODE REVIEW

**Comment by avsm** [comment]
oops this might supersede #547 @samoht -- also includes 0.11 improvements

**Comment by talex5** [comment]
The Docker files don't build. Do they work for you? Maybe they need a newer base image...

```
The following dependencies couldn't be met:
  - protocol-9p -> lwt = 3.0.0
    no matching version
```

**Comment by avsm** [comment]
I've pushed a working refreshed base image and also fixed the tests.

**Comment by avsm** [comment]
I dont understand why CircleCI failed...

**Comment by samoht** [comment]
'cause you need the other https://github.com/moby/datakit/pull/547/commits/0bf8e879a44c8b5e44fc720e943829eb1e0ea984
(and the other patches might be useful too)

**Comment by avsm** [comment]
thx @samoht, cherry picked your patches

**Comment by samoht** [comment]
I've pushed a fix for appveyor

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 #FROM ocaml/opam-dev:alpine-3.5_ocaml-4.04.0
-FROM ocaml/opam-dev@sha256:337fd0f78b182a12da3f66d9d5ca075dbb79df414fc01d6a0aee99ffeb35552a
 ENV OPAMERRLOGLEN=0 OPAMYES=1
 RUN sudo apk add tzdata aspcud gmp-dev perl
 
 RUN opam pin add hvsock.0.14.0 -n https://github.com/mirage/ocaml-hvsock.git#88397c37a910c61507a9655ed8413c1692294da0
 RUN opam pin add irmin.1.1.0 -n https://github.com/mirage/irmin.git#89196ad17c53b02f333022a87ecc264ec8c06af0
 RUN opam pin add irmin-git.1.1.0 -n https://github.com/mirage/irmin.git#89196ad17c53b02f333022a87ecc264ec8c06af0
-RUN opam pin add -yn protocol-9p 'https://github.com/talex5/ocaml-9p.git#ping-mirage-3'
+RUN opam pin add -yn protocol-9p.0.11.0 'https://github.com/mirage/ocaml-9p.git#v0.11.0'
```
Comment (samoht) [inline Dockerfile]: @talex5 @djs55 do we still need the fork? I think 0.11.0 has all the patches needed?

**Pull request merged**